src/lib.rs

```diff
@@ -99,33 +124,67 @@ pub fn run(args: Args) {
             );
 
             get_searcher(output_mode)
+                .borrow_mut()
                 .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
                 .unwrap();
+            buffer_writer.print(printer.get_mut()).unwrap();
         });
 }
 
-fn format_relative_path(path: &Path, output_mode: OutputMode) -> &Path {
-    if output_mode == OutputMode::Vimgrep && path.starts_with("./") {
-        path.strip_prefix("./").unwrap()
-    } else {
-        path
-    }
+type Printer = Standard<Buffer>;
+
+thread_local! {
+    static PRINTER: OnceCell<(Rc<RefCell<Printer>>, OutputMode)> = Default::default();
```

Comment:
Given wanting to (mimiicking my understanding of what `ripgrep` was doing via the `WalkParallel.run()` callback) have a single printer/searcher per thread, I reached for a `thread_local!` + `OnceCell` + `Rc`-wrapper (since a reference to the `thread_local!` value can't escape its `.with()` callback, so you'd either have to re-arrange the code using the printer to be expressed as inside such a callback or else you can use eg an `Rc`-wrapper to let a reference "leak" outside of the `.with()` callback) + `RefCell` inner wrapper (since we need to call `&mut self` methods on the printer/searcher, and inside an `Rc` you need interior mutability in order to do that)